source/Calamari.Tests/Fixtures/Deployment/DeployPackageWithStructuredConfigurationFixture.cs

```diff
@@ -311,6 +300,150 @@ public void FailsAndWarnsIfAFileCannotBeParsed()
+                Variables.Set(ActionVariables.StructuredConfigurationVariablesTargets, XmlFileNameWithNonXmlExtension);
+                Variables.Set("/document/key", "new-value");
+
+                var result = DeployPackage(file.FilePath);
+                result.AssertSuccess();
+
+                var extractedPackageUpdatedXmlFile = File.ReadAllText(Path.Combine(StagingDirectory, ServiceName, ServiceVersion, XmlFileNameWithNonXmlExtension));
+
+                // NOSHIP: Assert that it logs that it tried json first
+                this.Assent(extractedPackageUpdatedXmlFile, TestEnvironment.AssentXmlConfiguration);
+            }
+        }
+
+        [Test]
+        public void ShouldPerformReplacementInYamlFileWithANonYamlExtension()
+        {
+            using (var file = new TemporaryFile(PackageBuilder.BuildSamplePackage(ServiceName, ServiceVersion)))
+            {
+                Variables.Set(KnownVariables.Package.EnabledFeatures, KnownVariables.Features.StructuredConfigurationVariables);
+                Variables.Set(ActionVariables.StructuredConfigurationVariablesTargets, YamlFileNameWithNonYamlExtension);
+                Variables.Set("key", "new-value");
+
+                var result = DeployPackage(file.FilePath);
+                result.AssertSuccess();
+
+                var extractedPackageUpdatedYamlFile = File.ReadAllText(Path.Combine(StagingDirectory, ServiceName, ServiceVersion, YamlFileNameWithNonYamlExtension));
+
+                // noship: assert that it tried json, xml and json first
+                this.Assent(extractedPackageUpdatedYamlFile, TestEnvironment.AssentYamlConfiguration);
+            }
+        }
+
+        [Test]
+        public void ShouldPerformReplacementInPropertiesFileWithANonPropertiesExtension()
+        {
+            using (var file = new TemporaryFile(PackageBuilder.BuildSamplePackage(ServiceName, ServiceVersion)))
+            {
+                Variables.Set(KnownVariables.Package.EnabledFeatures, KnownVariables.Features.StructuredConfigurationVariables);
+                Variables.Set(ActionVariables.StructuredConfigurationVariablesTargets, PropertiesFileNameWithNonPropertiesExtension);
+                Variables.Set("debug", "false");
+                Variables.Set("port", "80");
+
+                var result = DeployPackage(file.FilePath);
+                result.AssertSuccess();
+
+                var extractedPackageUpdatedPropertiesFile = File.ReadAllText(Path.Combine(StagingDirectory, ServiceName, ServiceVersion, PropertiesFileNameWithNonPropertiesExtension));
+
+                // noship: assert that it tried json, xml and yaml first
+                this.Assent(extractedPackageUpdatedPropertiesFile, TestEnvironment.AssentPropertiesConfiguration);
+            }
+        }
+
+        [Test]
+        public void FailsAndWarnsIfFileWithAnXmlExtensionContainsValidNonXmlContent()
```

Comment:
This (and similar tests below) test that if a file has a well known extension, for example `.xml`, then we _don't_ try other parsers (like yaml, properties, json) even if the content was not parseable as `xml` content